README.md

```diff
@@ -136,11 +146,26 @@ SELECT email FROM mytable WHERE name = 'Evil Bob'
 | evilbob@gmail.com |
 +-------------------+
 ```
+## Custom data source implementation
+
+To be able to create your own data source implementation you need to implement the following interfaces:
+
+- `sql.Database` interface. This interface will provide tables from your data source.
+  - If your database implementation supports add more tables, you might want to add support to `sql.Alterable` interface
+
+- `sql.Table` interface. It will be in charge to transform any kind of data to an iterator of Rows. Depending of how much you want to optimize the queries, you also can implement other interfaces on your tables:
+  - `sql.PushdownProjectionTable` interface will provides you a way to get only the columns needed for the executed query.
+  - `sql.PushdownProjectionAndFiltersTable` interface will provide the same functionality described before, but also will push down the filters used in the executed query. Doing this, it allows you to filter before hand data, and improve query speed.
```

Comment:
s/it allows you to/you can/
s/before hand data/data beforehand/
s/query speed/the query speed/

Reply:
Maybe:
>> 
It allows to filter a data in advance, and speed up query
>>